### ISSUE TITLE
 [Critical bug] RabbitMQ healthchecks spawns new rmq connection each time

### ISSUE THREAD

**Issue by sergeyshaykhullin** [Xabaril/AspNetCore.Diagnostics.HealthChecks#563]
**What happened**:
When i used
```
.AddRabbitMQ(sp => sp.GetRequiredService<IConnectionFactory>())
.AddRabbitMQ(sp => sp.GetRequiredService<IConnectionFactory>().CreateConnection())
```
extensions methods, each time healthcheck is running new connection is spawned

**What you expected to happen**: 
HealthCheck Factory calls each check, but connection created once

**How to reproduce it (as minimally and precisely as possible)**:
```
.AddRabbitMQ(sp => sp.GetRequiredService<IConnectionFactory>())
.AddRabbitMQ(sp => sp.GetRequiredService<IConnectionFactory>().CreateConnection())
```

This is happening because `DefaultHealthCheckService` is calling `HealthCheckRegistration.Factory` each healthcheck run, but when you are using extension methods above - you are not using `(_) => instance` factory, but passed as argument with connection creation on each call

**Anything else we need to know?**:
This bug is causing memory leak, i found workaround registering IConnection as a singleton and passing in a factory, but for now this 2 extension methods are deadly!

**Environment**:
- Healthchecks version: RMQ - 3.1.2

**Comment by sergeyshaykhullin** [comment]
Also using empty `.AddRabbitMQ()` with IConnectionFactory registration spawning infinite connections

**Comment by unaizorrilla** [comment]
Hi @sergeyshaykhullin 

Can you see the examples [here](https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/a573cd2cf9bc09d8d0f790d9b54ff5ed94dca783/src/HealthChecks.RabbitMQ/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs#L139) ?

You can use the overload https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/a573cd2cf9bc09d8d0f790d9b54ff5ed94dca783/src/HealthChecks.RabbitMQ/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs#L74

This reuse the registered IConnection or IConnectionFactory on IServiceCollection, and normally you registered the singleton IConnectionFactory ( please see the examples )

Also the overloads

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/a573cd2cf9bc09d8d0f790d9b54ff5ed94dca783/src/HealthChecks.RabbitMQ/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs#L115


and

https://github.com/Xabaril/AspNetCore.Diagnostics.HealthChecks/blob/a573cd2cf9bc09d8d0f790d9b54ff5ed94dca783/src/HealthChecks.RabbitMQ/DependencyInjection/RabbitMQHealthCheckBuilderExtensions.cs#L139

allow you to set the factory to get this ( IConnection and IConnectionFactory ) and you can set this as Singleton in you IServiceCollection

**Comment by unaizorrilla** [comment]
Closed with #587
